src/Chapter2.hs

```diff
@@ -838,11 +856,22 @@ list.
 >>> rotate 1 [1,2,3,4]
 [2,3,4,1]
 >>> rotate 3 [1,2,3,4]
-[4,1,2,3]
+[2,1,3,4]
```

Comment:
This should not change, the correct solution shoud be `[4,1,2,3]` for the rotation. I see that you used different rules for the function. You can share what this function does in your case and I will then leave the feedback for that. Or you can try to implement the original task 🙂